packages/firebase_ml/android/src/main/java/io/flutter/plugins/firebaseml/FirebaseMLPlugin.java

```diff
@@ -40,21 +50,86 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBindin
     channel.setMethodCallHandler(this);
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.N)
   @Override
-  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
+  public void onMethodCall(@NonNull MethodCall call, @NonNull final Result result) {
+    final FirebaseCustomRemoteModel remoteModel;
+    String modelName;
+
     switch (call.method) {
-      case "downloadRemoteModel":
-        result.success("Testing call. Will be removed shortly");
+      case "FirebaseModelManager#download":
+        assert (call.argument("model") != null);
+        assert (call.argument("conditions") != null);
+
+        modelName = call.argument("model");
+        Map<String, Boolean> conditionsToMap = call.argument("conditions");
+        remoteModel = new FirebaseCustomRemoteModel.Builder(modelName).build();
+        FirebaseModelDownloadConditions.Builder conditionsBuilder =
+            new FirebaseModelDownloadConditions.Builder();
+        if (conditionsToMap.get("requireCharging")) conditionsBuilder.requireCharging();
+        if (conditionsToMap.get("requireDeviceIdle")) conditionsBuilder.requireDeviceIdle();
+        if (conditionsToMap.get("requireWifi")) conditionsBuilder.requireWifi();
+
+        FirebaseModelDownloadConditions conditions = conditionsBuilder.build();
+        FirebaseModelManager.getInstance()
+            .download(remoteModel, conditions)
+            .addOnCompleteListener(
+                new OnCompleteListener<Void>() {
+                  @Override
+                  public void onComplete(@NonNull Task<Void> task) {
+                    result.success(remoteModelToMap(remoteModel));
+                  }
+                });
+        break;
+      case "FirebaseModelManager#getLatestModelFile":
+        assert (call.argument("model") != null);
+
+        modelName = call.argument("model");
+
+        remoteModel = new FirebaseCustomRemoteModel.Builder(modelName).build();
+
+        FirebaseModelManager.getInstance()
+            .getLatestModelFile(remoteModel)
+            .addOnCompleteListener(
+                new OnCompleteListener<File>() {
+                  @Override
+                  public void onComplete(@NonNull Task<File> task) {
+                    File modelFile = task.getResult();
+                    if (modelFile != null) result.success(modelFile.getAbsolutePath());
+                    else task.getException().printStackTrace();
+                  }
+                });
```

Comment:
optional: I'm not sure if you are building on java 8, but if so you may optionally replace the anonymous inner class with a lambda.

Reply:
It does throw me an error. I would prefer to keep it to be consistent with other implementations of plugins.